### PR TITLE
MNT Remove js prefix from require statements

### PR DESCRIPTION
**Pull request by emteknetnz**
eslint-config PR
- [ ] https://github.com/silverstripe/eslint-config/pull/8

Related:
- [ ] https://github.com/silverstripe/silverstripe-asset-admin/pull/1238
- [ ] https://github.com/silverstripe/silverstripe-campaign-admin/pull/212
- [ ] https://github.com/silverstripe/silverstripe-ckan-registry/pull/256
- [ ] https://github.com/silverstripe/silverstripe-cms/pull/2700
- [ ] https://github.com/silverstripe/silverstripe-tagfield/pull/196

### CODE REVIEW

File: client/src/bundles/vendor.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ require('json-js');
 
 // jQuery plugins require that the jQuery object is exposed as a global
 // webpack.ProvidePlugin is used to ensure that jQuery and $ are provided to all includes
-require('script-loader!../../../thirdparty/jquery/jquery.js');
+require('script-loader!../../../thirdparty/jquery/jquerys');
```
Comment (maxime-rainville) [inline client/src/bundles/vendor.js:14]: I think there's a typo in the statement.

```suggestion
require('script-loader!../../../thirdparty/jquery/jquery');
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: client/src/bundles/vendor.js (hunk 1 above)
Comment (emteknetnz) [reply client/src/bundles/vendor.js]: Ha fixed

File: client/src/lib/tests/ShortcodeSerialiser-test.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /* global jest, jasmine, describe, beforeEach, it, pit, expect, process */
-
+/* eslint-disable
+ import/extensions
```
Comment (maxime-rainville) [inline client/src/lib/tests/ShortcodeSerialiser-test.js:3]: It's not clear to me why this test can't use a normal `import` statement or why we need a `dontMock` statement here.

File: client/src/lib/tests/router-test.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 /* global jest, jasmine, describe, beforeEach, it, expect, process */
-
+/* eslint-disable
+ import/extensions
+*/
 jest.dontMock('../Router.js');
```
Comment (maxime-rainville) [inline client/src/lib/tests/router-test.js:5]: Maybe try a normal `import` statement and/or remove the `dontMock` statement

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
When we are disabling a single thing, I think it would be better to use `// eslint-disable-next-line no-alert`.

Some of those `require` calls look weird. Before we go ahead and add linting exceptions, I would just like us to confirm that the weirdness is actually necessary.

File: client/src/boot/BootRoutes.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 /* global window, document */
-
+/* eslint-disable
+ no-alert
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/boot/BootRoutes.js:3]: Updated

File: client/src/legacy/DateField.js [hunk 5]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable
+ import/extensions
+*/
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/legacy/DateField.js:3]: Doesn't seem like it's required, removed it, ran `yarn build` and I can still filter SiteTree's using the DateField

File: client/src/lib/tests/ShortcodeSerialiser-test.js (hunk 2 above)
Comment (emteknetnz) [reply client/src/lib/tests/ShortcodeSerialiser-test.js:3]: update,d works as an import statement

File: client/src/lib/tests/router-test.js (hunk 3 above)
Comment (emteknetnz) [reply client/src/lib/tests/router-test.js:5]: update,d works as an import statement

**Comment by emteknetnz** [comment]
@maxime-rainville have made changed

> // eslint-disable-next-line no-alert.

I've retained the file-wide disables consistency with the other rules we ignore file wide e.g. `import/no-unresolved`

**Review by maxime-rainville — APPROVED** [review]
Maxime is happy

**Pull request merged**